### PR TITLE
Abandoned plugin support ( Action Hook)

### PR DESCRIPTION
**Pull request by angbpy**

### CODE REVIEW

File: includes/api/save-abandonment-data.php [hunk 1]
```diff
@@ -59,6 +60,8 @@ function saveCartAbandonmentData(WP_REST_Request $request)
     $result['response']    = "";
     $result['status_code'] = 400;
 
+    abandonedPluginHook($razorpayData); // do_action to notify/send the abandonedCart data to 3rd party plugins
```
Comment (ChetanGN) [inline includes/api/save-abandonment-data.php:63]: This function should call right after the try-catch block if there is no other dependency.

File: includes/support/abandoned-cart-hooks.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+     $rzpAbandonedData                                = array();
+     $rzpAbandonedData['woocommerceOrderID']          = $razorpayData['receipt'];
+     $rzpAbandonedData['amount']                      = $razorpayData['amount'];
+     $rzpAbandonedData['currency']                    = $razorpayData['currency'];
+     $rzpAbandonedData['customer_details']            = array();
+     $rzpAbandonedData['customer_details']['contact'] = $razorpayData['customer_details']['contact'];
+     $rzpAbandonedData['customer_details']['email']   = $razorpayData['customer_details']['email'];
+
+    $shippingAddress                                  = array(
+        'name'     => $razorpayData['customer_details']['shipping_address']['name'],
```
Comment (ChetanGN) [inline includes/support/abandoned-cart-hooks.php:14]: Define the variable with the shipping address like
`$magicShippingAddress = $razorpayData['customer_details']['shipping_address']` and use this here. Same for the billing address.

File: includes/support/abandoned-cart-hooks.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+function abandonedPluginHook($razorpayData) {
+
+     $rzpAbandonedData                                = array();
+     $rzpAbandonedData['woocommerceOrderID']          = $razorpayData['receipt'];
+     $rzpAbandonedData['amount']                      = $razorpayData['amount'];
+     $rzpAbandonedData['currency']                    = $razorpayData['currency'];
+     $rzpAbandonedData['customer_details']            = array();
+     $rzpAbandonedData['customer_details']['contact'] = $razorpayData['customer_details']['contact'];
+     $rzpAbandonedData['customer_details']['email']   = $razorpayData['customer_details']['email'];
```
Comment (ChetanGN) [inline includes/support/abandoned-cart-hooks.php:11]: As email will be an optional field in the modal, will that affect the flow? varify this @angbpy

**Review by ChetanGN — CHANGES_REQUESTED** [review]

File: includes/support/abandoned-cart-hooks.php (hunk 3 above)
Comment (angbpy) [reply includes/support/abandoned-cart-hooks.php:11]: This shouldn't affect the flow as email empty/not-empty checks are performed at the 3rd party plugin end before insertion

Comment (ChetanGN) [reply includes/support/abandoned-cart-hooks.php:11]: What if the email key does not exist in rzp object? verify this flow.

**Review by poojarazor — APPROVED** [review]
LGTM

**Review by angbpy — COMMENTED** [review]
Changes already made

**Review by ChetanGN — APPROVED** [review]

**Pull request merged**